### PR TITLE
completed string-manipulation-2 | Daniel Hong

### PR DESCRIPTION
**Pull request by daniel-sj-hong**
Capture

### CODE REVIEW

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Hey dude! Please resubmit with your pseudocode included for each function you defined

File: string-manipulation-2/src/first-chars.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  var container = '';
+  for (var i = 0; i < length; i++) {
+    if (string[i] === undefined) {
+      break;
+    }
+    container += string[i];
+  }
+  if (string === '') {
+    return container;
+  }
```
Comment (thebearingedge) [inline string-manipulation-2/src/first-chars.js:12]: I recommend doing this right after `container` is declared. It's good to get the special cases like this out of the way near the top of the function

File: string-manipulation-2/src/last-chars.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* exported lastChars */
+function lastChars(length, string) {
+  var container = '';
+  for (var i = string.length - length; i < string.length; i++) {
+    container += string[i];
+  }
+  if (length > string.length) {
+    return string;
+  }
```
Comment (thebearingedge) [inline string-manipulation-2/src/last-chars.js:9]: Same here

File: string-manipulation-2/src/truncate.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  var container = '';
+  for (var i = 0; i < length; i++) {
+    if (string[i] === undefined) {
+      break;
+    }
+    container += string[i];
+  }
+  if (string === '') {
+    return '...';
+  }
```
Comment (thebearingedge) [inline string-manipulation-2/src/truncate.js:12]: And here

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**